### PR TITLE
#96 Added Ingame Assembler Functions

### PR DESCRIPTION
**Pull request by mrsib**
Added Ingame Assembler functions QueueItem and ClearQueue to enable
players to be able to queue items in the assembler and clear the queue.
QueueItem has three parameters, string itemType (I.e.
MyObjectBuilder_Component), string subtypeName(I.e. SteelPlate), int
amount
QueueItem returns bool: true if item was queued, false if not. False
indicates invalid type or subtype name.

ClearItem has no parameters

Both functions work in either assemble mode or disassemble mode.

### CODE REVIEW

File: Sources/Sandbox.Game/Game/Entities/Blocks/MyAssembler.cs [hunk 1]
```diff
@@ -1065,5 +1065,22 @@ protected override float GetOperationalPowerConsumption()
...
+
+        bool Sandbox.ModAPI.Ingame.IMyAssembler.QueueItem(string itemType, string subtypeName, int amount)
+        {
+            try
+            {
+                InsertQueueItemRequest(-1,
+                                       MyDefinitionManager.Static.TryGetBlueprintDefinitionByResultId(new MyDefinitionId(MyObjectBuilderType.Parse(itemType), subtypeName)),
+                                       (VRage.MyFixedPoint)amount);
+            }
+            catch { return false; } // User probably entered invalid type or name
```
Comment (OndrejPetrzilka) [inline Sources/Sandbox.Game/Game/Entities/Blocks/MyAssembler.cs]: I would prefer something else than this. Ideally if checks.

Comment (mrsib) [inline Sources/Sandbox.Game/Game/Entities/Blocks/MyAssembler.cs]: Understood, will look into further.

**Comment by mrsib** [comment]
Closing for further alterations, will make new pull request when ready.

**Comment by mexmer** [comment]
since issue is dicussed here #96 i would like to see better approach to this issue.

**Pull request closed without merging**